src/main/java/seedu/address/logic/LogicManager.java

```diff
@@ -46,6 +48,7 @@ public LogicManager(Model model, Storage storage) {
     public CommandResult execute(String commandText) throws CommandException, ParseException {
         logger.info("----------------[USER COMMAND][" + commandText + "]");
 
+        HistoryCommandList.setCommandHistoryText(commandText);
```

Comment:
instead of passing the last `commandText` in this manner, then constructing a new `HistoryCommand` inside each of the add/clear/edit/delete commands, how about this:

- do an `if` check inside this function to see if it is one of add/clear/edit/delete
- call `.add()` here with the `commandText`

this way we only pass `commandText` when we need it. also, if we want to know exactly which `Command` it is, we can do that check here (but i'm not sure if we actually need it?)

Reply:
then this section of code can also help to save the state of the list in the future